### PR TITLE
chore: update tracing and logging docs

### PR DESCRIPTION
**Pull request by tustvold**
Still pretty barebones, but these docs were very out of date, and I could see the potential for much confusion

### CODE REVIEW

File: docs/logging.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+fn say_hello(name: &str) {
+	// This is good - fields come first!
+    info!(name, "hello there");
+
+	// This is alright, but less structured / useful
+    info!("hello there {}", name);
+}
+```
+
+## Filtering
```
Comment (alamb) [inline docs/logging.md:24]: If you wanted to make these docs even better I would recommend
1. Calling this document "Developers Guide To Logging"
2. Add another doc (or perhaps section at the top of this one) called "Users Guide to Tracing" that talks about `LOG_FILTER` and `--log-filter` and provides some examples

And include

File: docs/tracing.md [hunk 2]
```diff
@@ -1,163 +1,110 @@
-# Tracing in IOx
+# Distributed Tracing in IOx
 
-IOx makes use of Rust's tracing ecosystem to output both application logs and
-tracing events to Jaeger for distributed request correlation.
+Not to be confused with the [tracing](https://docs.rs/tracing) ecosystem, which we use for [logging](logging.md),
```
Comment (alamb) [inline docs/tracing.md:3]: 🤦  so sad the `tracing` crate is used for logging

Comment (alamb) [reply docs/tracing.md:3]:
```suggestion
Not to be confused with the Rust [tracing](https://docs.rs/tracing) ecosystem, which we use for [logging](logging.md),
```

File: docs/tracing.md [hunk 3]
```diff
@@ -1,163 +1,110 @@
...
+## Running Jaeger / tracing locally
 
-First off, when running IOx you need to choose how much information you want.
-You do this by setting the `RUST_LOG="debug"` environment variable (or `info`,
-`error`, etc) or passing `-v` or `-vv` on the command line. The `RUST_LOG`
-environment variable provides [granular control] over what log verbosity is
-configured for individual internal components (tip: try `RUST_LOG="iox=debug"`!)
+IOx fits into the distributed tracing ecosystem, as described in
+[tracing](docs/tracing.md). To use, develop, or debug the distributed tracing functionality locally you can do the
+following:
```
Comment (alamb) [inline docs/tracing.md:54]:
```suggestion
 To use, develop, or debug the distributed tracing functionality locally you can do the
following:
```

(I don't think there is much value for a link back to itself :) )

**Review by alamb — APPROVED** [review]

File: docs/logging.md (hunk 1 above)
Comment (tustvold) [reply docs/logging.md]: I added a "Basic Usage" section at the top, and moved the rest into a section called "Developer Guide"

**Pull request merged**